src/social-card/insight-card/insight-card.service.ts

```diff
@@ -0,0 +1,168 @@
+import { ForbiddenException, Injectable, Logger, NotFoundException } from "@nestjs/common";
+import { HttpService } from "@nestjs/axios";
+import { Resvg } from "@resvg/resvg-js";
+import fs from "node:fs/promises";
+import { GithubService } from "../../github/github.service";
+import { S3FileStorageService } from "../../s3-file-storage/s3-file-storage.service";
+
+import tailwindConfig from "../templates/tailwind.config";
+import { firstValueFrom } from "rxjs";
+
+import { RequiresUpdateMeta } from "../user-card/user-card.service";
+import { DbInsight } from "../../github/entities/db-insight.entity";
+import insightCardTemplate from "../templates/insight-card.template";
+import insightRepos from "../templates/shared/insight-repos";
+import insightContributors from "../templates/shared/insight-contributors";
+
+interface InsightCardData {
+  pageName: string;
+  repos: { repoName: string; avatarUrl: string }[];
+  contributors: string[];
+  updated_at: Date;
+}
+
+@Injectable()
+export class InsightCardService {
+  private readonly logger = new Logger(this.constructor.name);
+
+  constructor (
+    private readonly httpService: HttpService,
+    private readonly githubService: GithubService,
+    private readonly s3FileStorageService: S3FileStorageService,
+  ) {}
+
+  private async getInsightData (insightId: number): Promise<InsightCardData> {
+    const insightPageReq = await firstValueFrom(
+      this.httpService.get<DbInsight>(`https://api.opensauced.pizza/v1/insights/${insightId}`),
+    );
+
+    const { repos, name, updated_at } = insightPageReq.data;
+
+    const query = (new URLSearchParams);
+
+    query.set("repoIds", repos.map(repo => repo.repo_id).join(","));
```

Comment:
We should probably limit the number of repos passed to the URL. 
```suggestion
    query.set("repoIds", repos.map(repo => repo.repo_id).join(","));
```

Reply:
Like max 10?